Sources/VaporToolbox/PrepareAdd.swift

```diff
@@ -0,0 +1,78 @@
+import Foundation
+import Console
+
+public final class PrepareAdd: Command {
+    public let id = "add"
+
+    public let signature: [Argument] = []
+
+    public let help: [String] = [
+        "Create new database preparation file."
+    ]
+
+    public let console: ConsoleProtocol
+
+    public init(console: ConsoleProtocol) {
+        self.console = console
+    }
+
+    public func run(arguments: [String]) throws {
+        let preparationsFolder = "./Sources/App/Preparations"
+
+        do {
+            _ = try console.backgroundExecute(program: "ls", arguments: ["./Sources/App/main.swift"])
+        } catch ConsoleError.backgroundExecute(_) {
+            throw ToolboxError.general("Invalid Vapor template: could not find Sources/App/main.swift file")
+        }
+
+        do {
+            _ = try console.backgroundExecute(program: "ls", arguments: [preparationsFolder])
+        } catch ConsoleError.backgroundExecute(_) {
+            console.warning("No preparations folder found (\(preparationsFolder)). Creating now...")
+        }
+
+        do {
+            _ = try console.backgroundExecute(program: "mkdir", arguments: ["-p", preparationsFolder])
+        } catch ConsoleError.backgroundExecute(_) {
+            throw ToolboxError.general("Failed to create preparations folder (\(preparationsFolder)).")
+        }
+
+        if arguments.count < 2 {
```

Comment:
Should this error possibly be higher in the order of events so we can fail right away?